compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/FirSourceChild.kt

```diff
@@ -36,5 +36,12 @@ private fun FirPsiSourceElement<*>.getChild(types: Set<IElementType>, index: Int
 
 private fun FirLightSourceElement.getChild(types: Set<IElementType>, index: Int, depth: Int): FirSourceElement? {
     val visitor = LighterTreeElementFinderByType(treeStructure, types, index, depth)
-    return visitor.find(lighterASTNode)?.toFirLightSourceElement(treeStructure)
+    val node = lighterASTNode
+    val childNode = visitor.find(node) ?: return null
+    val offset = startOffset - node.startOffset
```

Comment:
It would be helpful if some comments with examples are added around here.

Reply:
Huh, this morning when I try again it looks like offset is now always 0. Yesterday I added this because for org.jetbrains.kotlin.test.runners.FirDiagnosticsWithLightTreeTestGenerated.Resolve.Arrays#testArraySetWithOperation, I am getting `lighterASTNode.startOffset == 6` and `this.startOffset > 500`. Then the obtained child node also has start offset as 6, which then causes the test infra to place the diagnostic on the very top of that test file, among the comments in the first line. 6 is a weird number since it's definitely not the relative position from the immediate parent. I am guessing it's the position from some partial tree? But I don't know when a partial tree is used in this case since this morning I am no longer able to reproduce the behavior.